pages/films/index.tsx

```diff
@@ -0,0 +1,276 @@
+import "./index.scss"
+import Paginator from "../../components/base/Paginator/Paginator"
+import SearchField from "../../components/base/SearchField/SearchField"
+import { observer } from "mobx-react-lite"
+import { initializeStore, useStore } from "../../store/store"
+import React, { useEffect, useMemo } from "react"
+import { useRouter } from "next/router"
+import PageSizeSelector from "../../components/base/PageSizeSelector/PageSizeSelector"
+import { NextPageContext } from "next"
+import { toJS } from "mobx"
+import FilmsOrderBySelector from "../../components/films/FilmsOrderBySelector/FilmsOrderBySelector"
+import SortDirectionToggler from "../../components/base/SortDirectionToggler/SortDirectionToggler"
+import Sidebar from "../../components/base/Sidebar/Sidebar"
+import Header from "../../components/base/Header/Header"
+import FilmCard from "../../components/films/FilmCard/FilmCard"
+
+const defaultTotalItemsCount = 10
+const defaultCurrentPage = 1
+const defaultFilmDate = "1"
+
+const Search = observer(() => {
+  const store = useStore()
+  const route = useRouter()
+
+  type handleChangeQueryType = {
+    page?: string
+    perPage?: string
+    filmTitle?: string
+    type?: string
+    date_range?: string
+    order_by?: string
+    sort?: string
+  }
+
+  const handleChangeQuery = ({
+    page,
+    perPage,
+    filmTitle,
+    type,
+    date_range,
+    order_by,
+    sort,
+  }: handleChangeQueryType) => {
+    route.push({
+      query: {
+        // Сетаем параметры из URL,
+        // прежде чем сетать новые параметры, чтобы не терять предыдущее значение
+        ...route.query,
+        ...(page ? { page } : {}),
+        ...(perPage ? { perPage } : {}),
+        ...(filmTitle ? { filmTitle } : {}),
+        ...(type ? { type } : {}),
+        ...(date_range ? { date_range } : {}),
+        ...(order_by ? { order_by } : {}),
+        ...(sort ? { sort } : {}),
+      },
```

Comment:
здесь нужно заюзать функцию, которую ты делал
```suggestion
      query: {
        // Сетаем параметры из URL,
        // прежде чем сетать новые параметры, чтобы не терять предыдущее значение
        ...route.query,
        ...assignIfDefined({
          page,
          perPage,
          filmTitle,
          type,
          date_range,
          order_by,
          sort
        }),
      },
```